src/service/newsletter/newsletter-subscription-service.ts

```diff
@@ -66,12 +66,17 @@ export class NewsletterSubscriptionService {
     return response.data;
   }
 
-  async unsubscribeFromNewsletter(
-    plaintextEmail: string,
-    type: NewsletterType,
-    unsubscribeReferenceId?: string,
-    sub?: string,
-  ): Promise<void> {
+  async unsubscribeFromNewsletter({
+    plaintextEmail,
+    type,
+    unsubscribeReferenceId,
```

Comment:
If a user unsubscribes via the front-end this value is not going to be populated. We need some logic in here to get the unsubscribe reference by the user id and newsletter type and use that to delete the reference.

Reply:
aye I've got that in here: https://github.com/cabinetoffice/gap-find-backend/pull/69